dkeyserver/App/main.cpp

```diff
@@ -174,6 +215,17 @@ int main()
         return -1;
     }
 
+    if (target_ip_addr.empty() || target_port == 0)
+    {
+        log_d("no target server. \n");
+    }else{
+        ret = sgx_confirm_target(g_enclave_id, &ret, target_ip_addr.c_str(), target_port);
+        if (ret != 0)
+        {
+            log_i("failed to connect the target server.\n");
```

Comment:
here need to return directly.

Reply:
yes, return error directly

Reply:
and seems there're a lot of duplicated codes with the sgx_set_up_tls_server and sgx_confirm_target, can we refine it?

seems that you only need to re-implement the sgx_get_domainkey(), to check whether retrieve the dk from disk or the target server

Reply:
done.